src/tds/read.c

```diff
@@ -240,8 +240,12 @@ tds_get_n(TDSSOCKET * tds, void *dest, size_t need)
 			dest = (char *) dest + have;
 		}
 		need -= have;
-		if (TDS_UNLIKELY(tds_read_packet(tds) < 0))
+		if (TDS_UNLIKELY(tds->recv_packet->capacity < 2
+				 ||  tds->in_buf[1] != 0
```

Comment:
Maybe you want to test only the final bit and not all the flags. It makes sense, if no more packets are expected data should not continue on next packet. Wondering if `tds_get_n` would be called for first bytes of the packet series. Not sure if this can even happen however. I don't think cancellation apply here.

Reply:
Ah, yeah, testing just the low-order bit is more appropriate, though I'm not sure how much difference it makes in practice.

The situation originally motivating this change was canceling blob retrieval, which at least in the case of the OpenServer instance I was using, could yield a truncated row response giving a blob's full length but containing only part of its body, leading to loss of synchronization that would otherwise typically culminate in hangs.  Perhaps the server should have acknowledged cancellation more clearly, but it did at least set that flag correctly.

Reply:
Interesting! So you are telling the way Open Server handles cancellation is different? Maybe we can support it.

Reply:
Yes, at least historically.  Maybe it's been fixed since, but a little extra caution here can't hurt.